### PR TITLE
`tr` should error out when if user provides more than 2 sets

### PR DESCRIPTION
**Pull request by hbina**
```shell
coreutils on  master [$?] is 📦 v0.0.6 via 🦀 v1.53.0 
❯ echo "test" | ./target/debug/coreutils tr 'f' 'f' 'f'
test

coreutils on  master [$?] is 📦 v0.0.6 via 🦀 v1.53.0 
❯ echo "hello" | tr 'f' 'f' 'f'
tr: extra operand ‘f’
Try 'tr --help' for more information.
```

Signed-off-by: Hanif Bin Ariffin <hanif.ariffin.4326@gmail.com>

### CODE REVIEW

File: src/uu/tr/src/tr.rs [hunk 1]
```diff
@@ -281,6 +281,15 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
         return 1;
     }
 
+    if sets.len() > 2 {
```
Comment (sylvestre) [inline src/uu/tr/src/tr.rs:284]: Did you try to use https://docs.rs/clap/2.33.3/clap/struct.Arg.html#method.max_values instead?

**Review by sylvestre — CHANGES_REQUESTED** [review]
please a test too

File: src/uu/tr/src/tr.rs (hunk 1 above)
Comment (hbina) [reply src/uu/tr/src/tr.rs:284]: Nope, I simply didn't know about it.

Comment (sylvestre) [reply src/uu/tr/src/tr.rs:284]: clap rocks :)
In most of the case, it is doing the work for you :)

https://github.com/uutils/coreutils/pull/2486/files#diff-4fb0272cb5f6ecf7830f62afa24ec988dc425fc3b8ac002c1808b18662533568R267-R273
can probably removed btw

Comment (hbina) [reply src/uu/tr/src/tr.rs]: that doesn't appear to work.
Screenshot from 2021-07-09 23-55-11

Comment (sylvestre) [reply src/uu/tr/src/tr.rs]: please don't share screenshot: 
* they are terrible for accessibility
* they cannot be searched
*

Comment (hbina) [reply src/uu/tr/src/tr.rs]: Do you know a way to cat specific lines of code? I could grep it of course, but its not reliable.

Comment (sylvestre) [reply src/uu/tr/src/tr.rs]: it seems you found the fix, right ? :)

File: src/uu/tr/src/tr.rs [hunk 2]
```diff
@@ -259,7 +259,7 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
     let squeeze_flag = matches.is_present(options::SQUEEZE);
     let truncate_flag = matches.is_present(options::TRUNCATE);
 
-    let sets: Vec<String> = match matches.values_of(options::SETS) {
+    let sets = match matches.values_of(options::SETS) {
         Some(v) => v.map(|v| v.to_string()).collect(),
         None => vec![],
     };
```
Comment (jhscheer) [inline src/uu/tr/src/tr.rs:265]:
```suggestion
    let sets: Vec<String> = matches
        .values_of(options::SETS)
        .map(|v| v.map(ToString::to_string).collect())
        .unwrap_or_default();
```

File: src/uu/tr/src/tr.rs (hunk 1 above)
Comment (hbina) [reply src/uu/tr/src/tr.rs]: Any idea why `min_values` does not work?

```shell
coreutils on  hbina-tr-return-err-when-extra-args [$!?] is 📦 v0.0.6 via 🦀 v1.53.0 
❯ git diff
diff --git a/src/uu/tr/src/tr.rs b/src/uu/tr/src/tr.rs
index 1a696f3b..0105c81a 100644
--- a/src/uu/tr/src/tr.rs
+++ b/src/uu/tr/src/tr.rs
@@ -263,15 +263,7 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
         Some(v) => v.map(|v| v.to_string()).collect(),
         None => vec![],
     };
-
-    if sets.is_empty() {
-        show_error!(
-            "missing operand\nTry `{} --help` for more information.",
-            executable!()
-        );
-        return 1;
-    }
-
+    
     if !(delete_flag || squeeze_flag) && sets.len() < 2 {
         show_error!(
             "missing operand after '{}'\nTry `{} --help` for more information.",
@@ -364,6 +356,7 @@ pub fn uu_app() -> App<'static, 'static> {
             Arg::with_name(options::SETS)
                 .multiple(true)
                 .takes_value(true)
+                .min_values(1)
                 .max_values(2),
         )
 }

coreutils on  hbina-tr-return-err-when-extra-args [$!?] is 📦 v0.0.6 via 🦀 v1.53.0 
❯ cargo run -- tr
    Finished dev [unoptimized + debuginfo] target(s) in 0.21s
     Running `target/debug/coreutils tr`
tr: thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /home/hbina/git/coreutils/src/uu/tr/src/tr.rs:270:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: src/uu/tr/src/tr.rs (hunk 2 above)
Comment (hbina) [reply src/uu/tr/src/tr.rs]: Applied.

**Pull request merged**